### PR TITLE
Fix seed form validation bugs

### PR DESCRIPTION
**Pull request by badnames**
## Basics

- [x] I added a line to [changelog.md](/doc/changelog.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed the introduction tour
- [ ] I wrote migrations in a way that they are compatible with already present data
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Review by 4ydan — APPROVED** [review]
Looks good to me. Issue looks fixed.

**Comment by markus2330** [comment]
44.4 and abc didn't work for me on pr.permaplant.net? I'll rebuild and check again.

**Comment by markus2330** [comment]
jenkins build please

File: frontend/src/config/i18n/de/seeds.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "placeholder_binomial_name": "Tomate",
     "placeholder_origin": "Daheim",
     "placeholder_taste": "nussig",
-    "placeholder_yield": "1",
+    "placeholder_yield": "Hoch",
     "placeholder_price": "2,99€",
     "placeholder_generation": "0",
     "error_price_must_be_number": "Der Preis muss eine Zahl sein. z.B. 2,99",
```
Comment (markus2330) [inline frontend/src/config/i18n/de/seeds.json:25]: I only get "please enter a number" and not this text? And the error can be improved:

```suggestion
    "error_price_must_be_number": "Entschuldigung, der Preis muss eine Zahl mit . als Komma sein. z.B. 2.99",

```

File: frontend/src/config/i18n/de/seeds.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "placeholder_binomial_name": "Tomate",
     "placeholder_origin": "Daheim",
     "placeholder_taste": "nussig",
-    "placeholder_yield": "1",
+    "placeholder_yield": "Hoch",
     "placeholder_price": "2,99€",
```
Comment (markus2330) [inline frontend/src/config/i18n/de/seeds.json:23]:
```suggestion
    "placeholder_price": "2.99€",

```

File: frontend/src/config/i18n/en/seeds.json [hunk 3]
```diff
@@ -19,7 +19,7 @@
     "placeholder_binomial_name": "Tomato",
     "placeholder_origin": "My Home",
     "placeholder_taste": "sweet",
-    "placeholder_yield": "1",
+    "placeholder_yield": "Good",
     "placeholder_price": "2,99€",
```
Comment (markus2330) [inline frontend/src/config/i18n/en/seeds.json:23]:
```suggestion
    "placeholder_price": "2.99€",

```

File: frontend/src/config/i18n/en/seeds.json [hunk 4]
```diff
@@ -19,7 +19,7 @@
     "placeholder_binomial_name": "Tomato",
     "placeholder_origin": "My Home",
     "placeholder_taste": "sweet",
-    "placeholder_yield": "1",
+    "placeholder_yield": "Good",
     "placeholder_price": "2,99€",
     "placeholder_generation": "0",
     "error_price_must_be_number": "Price must be a number",
```
Comment (markus2330) [inline frontend/src/config/i18n/en/seeds.json:25]:
```suggestion
    "error_price_must_be_number": "Sorry, price must be a number with . as comma, e.g. 2.99",
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Thank you for the fix! Please improve the error message. Can you also allow "," as comma? (okay if it is accepted for both German and English)

File: frontend/src/config/i18n/de/seeds.json (hunk 1 above)
Comment (markus2330) [reply frontend/src/config/i18n/de/seeds.json:25]: Or if we support ",", too:

```suggestion
    "error_price_must_be_number": "Entschuldigung, ich verstehe nur einen Preis als Zahl mit Punkt (.) oder Beistrich (,) als Komma, z.B. 2,99",
```

**Comment by badnames** [comment]
> Can you also allow "," as comma? (okay if it is accepted for both German and English)

https://stackoverflow.com/questions/26890209/html-tag-input-of-type-number-and-culture-based-decimal-separator

The comma seperator in HTML5 input fields is tied to the browsers language. 
We could change this using the 'lang' field of the root html tag, but I don't (yet) know if and how this can be set up using react and our internationalization library.

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
I still only get "please enter a number" and not the correct error text?

The error messages are now inconsistent, please fix them according what we decide in #895. In the current behavior you would need to say in both languages that it could be , or .

**Comment by badnames** [comment]
> I still only get "please enter a number" and not the correct error text?

> The error messages are now inconsistent, please fix them according what we decide in https://github.com/ElektraInitiative/PermaplanT/issues/895. In the current behavior you would need to say in both languages that it could be , or .

I have solved the original validation problem using the default html5 constraints for number fields (min, max, step, type, etc.) in the hopes that this would simplify the form a bit.
This means that the input validation for the price field is now primarily handled by the browser, which makes its behavior apparently a bit inconsistent.

On chrome it is for example not possible to input any characters besides numbers, while Firefox allows that kind of input and only displays an error message ("please enter a number") once the user tries to submit the form.

Furthermore, Chrome seems to select the comma separator according to the users System language (German in my case) while Firefox uses '.'.
Changing the language on the site (react i18n does apparently update the 'lang' attribute of the root element) has no consistent effect on the selected comma separator.


If you wish, I could make the input behavior consistent again by implementing custom validation logic.

File: frontend/src/config/i18n/en/seeds.json [hunk 5]
```diff
@@ -19,10 +19,11 @@
...
     "placeholder_origin": "My Home",
     "placeholder_taste": "sweet",
-    "placeholder_yield": "1",
-    "placeholder_price": "2,99€",
+    "placeholder_yield": "Good",
+    "placeholder_price": "2.99€",
+
     "placeholder_generation": "0",
-    "error_price_must_be_number": "Price must be a number",
+    "error_price_must_be_number": "Sorry, I only understand the price as a number with . as comma, e.g. 2.99",
```
Comment (markus2330) [inline frontend/src/config/i18n/en/seeds.json:26]:
```suggestion
    "error_price_must_be_number": "Sorry, I only understand the price as a number with dot (.) or comma (,) depending on your Browser's preferences as decimal separator, e.g. 2.99",
```

File: frontend/src/config/i18n/de/seeds.json [hunk 6]
```diff
@@ -19,10 +19,11 @@
...
     "placeholder_origin": "Daheim",
     "placeholder_taste": "nussig",
-    "placeholder_yield": "1",
-    "placeholder_price": "2,99€",
+    "placeholder_yield": "Hoch",
+    "placeholder_price": "2.99€",
+
     "placeholder_generation": "0",
-    "error_price_must_be_number": "Der Preis muss eine Zahl sein. z.B. 2,99",
+    "error_price_must_be_number": "Entschuldigung, ich verstehe nur Zahlen mit Punkt (.) oder Beistrich (,) als Komma, z.B. 2,99",
```
Comment (markus2330) [inline frontend/src/config/i18n/de/seeds.json:26]:
```suggestion
    "error_price_must_be_number": "Entschuldigung, ich verstehe nur Zahlen mit Punkt (.) oder Beistrich (,) abhängig von den Browsereinstellungen als Komma, z.B. 2,99",

```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Suggestions fix the inconsistent error messages.

**Comment by markus2330** [comment]
> I have solved the original validation problem using the default html5 constraints for number fields (min, max, step, type, etc.) in the hopes that this would keep the form nice and simple.
> This means that the input validation for the price field is now primarily handled by the browser, which makes its behavior apparently a bit inconsistent.

Please read my text carefully, I was talking about inconsistent "error messages". I now suggested how we can fix them so that I can finally merge this PR.

> On chrome it is for example not possible to input any characters besides numbers, while Firefox allows that kind of input and only displays an error message ("please enter a number") once the user tries to submit the form.

So you say this error message is hard-coded in Firefox and cannot be changed by us?

> If you wish, I could make the input behavior consistent again by implementing custom validation logic.

I don't want to work against the Browsers but rather with them. If we fix it, we fix in in a general way like https://github.com/ElektraInitiative/PermaplanT/issues/895.

**Comment by badnames** [comment]
> Please read my text carefully, I was talking about inconsistent "error messages". I now suggested how we can fix them so that I can finally merge this PR.

I know, this is just some background information on why these inconsistent error messages are displayed.

> So you say this error message is hard-coded in Firefox and cannot be changed by us?

I have not yet found any information suggesting otherwise.

**Review by markus2330 — APPROVED** [review]
Let us merge this, follow up #895 already exists.

**Pull request merged**